### PR TITLE
[OSDEV-644] Error when trying to delete a facility with only one contributor

### PR DESCRIPTION
**Pull request by roman-stolar**
[OSDEV-644](https://opensupplyhub.atlassian.net/browse/OSDEV-644) Error when trying to delete a facility with only one contributor
- Implemented logic to ensure that when we delete a facility, we also clear the corresponding data in the api_facilityclaimreviewnote table.
- Added tests

### CODE REVIEW

**Review by killev — APPROVED** [review]
Approved

File: src/django/api/views/facility/facilities_view_set.py [hunk 1]
```diff
@@ -909,7 +916,11 @@ def delete_facility_list_item(item):
                     delete_facility_match(other_match)
                     delete_facility_list_item(other_match.facility_list_item)
 
-        for claim in FacilityClaim.objects.filter(facility=facility):
+        claims = FacilityClaim.objects.filter(facility=facility)
+        for claim in claims:
+            delete_facility_claim_review_notes(claim)
```
Comment (vladsha-dev) [inline src/django/api/views/facility/facilities_view_set.py:921]: It is not efficient to execute two identical loops. I believe you can perform this action within the second loop.
```
        for claim in claims:
            delete_facility_claim_review_notes(claim)
            claim._change_reason = f'Deleted {facility.id}'
            claim.delete()
```

**Review by vladsha-dev — CHANGES_REQUESTED** [review]
Please, add a release note as well.

File: src/django/api/views/facility/facilities_view_set.py [hunk 2]
```diff
@@ -788,6 +789,12 @@ def delete_facility_list_item(item):
             item.facility = None
             item.save()
 
+        def delete_facility_claim_review_notes(claim):
+            notes = FacilityClaimReviewNote.objects.filter(claim=claim)
+            for claim_review_note in notes:
+                claim_review_note._change_reason = f'Deleted {facility.id}'
```
Comment (vladsha-dev) [inline src/django/api/views/facility/facilities_view_set.py:795]: Is it for the historical table?

**Review by VadimKovalenkoSNF — DISMISSED** [review]
Pls, update RELEASE-NOTES as well.

File: src/django/api/views/facility/facilities_view_set.py [hunk 3]
```diff
@@ -788,6 +789,12 @@ def delete_facility_list_item(item):
             item.facility = None
             item.save()
 
+        def delete_facility_claim_review_notes(claim):
```
Comment (vladsha-dev) [inline src/django/api/views/facility/facilities_view_set.py:792]: This function can be optimized.
```
def delete_facility_claim_review_notes(claim):
    FacilityClaimReviewNote.objects.filter(claim=claim).update(_change_reason=f'Deleted {claim.facility.id}')
    FacilityClaimReviewNote.objects.filter(claim=claim).delete()
```
By combining the update and delete operations into a single query, we can optimize the code and reduce the number of database queries executed.

**Review by vladsha-dev — CHANGES_REQUESTED** [review]

File: src/django/api/views/facility/facilities_view_set.py (hunk 2 above)
Comment (roman-stolar) [reply src/django/api/views/facility/facilities_view_set.py:795]: Yes.

**Review by vladsha-dev — APPROVED** [review]
LGTM

**Review by killev — APPROVED** [review]
LGTM. Approved

**Pull request merged**